diff --git a/dist/src/index.js b/dist/src/index.js
index dc526b90f24958bcdb9814642ad88f4e9c4873f5..ea7172e0cdc01ed676817917fa9dcafaca833ab7 100644
--- a/dist/src/index.js
+++ b/dist/src/index.js
@@ -519,7 +519,10 @@ function convertToByteStream(stream, clone = false) {
         controller.enqueue(result.value);
       }
     },
-    cancel: (reason) => reader.cancel(reason)
+    cancel: (reason) => {
+      if (reader === undefined) reader = stream.getReader();
+      return reader.cancel(reason);
+    }
   });
 }
 function convertToRegularStream(stream) {
@@ -539,7 +542,10 @@ function convertToRegularStream(stream) {
         controller.enqueue(result.value);
       }
     },
-    cancel: (reason) => reader.cancel(reason)
+    cancel: (reason) => {
+      if (reader === undefined) reader = stream.getReader();
+      return reader.cancel(reason);
+    }
   });
 }
 import_web2.ReadableStreamBYOBReader.prototype.readAtLeast = async function(bytes, view) {
@@ -799,7 +805,10 @@ var Body = class {
           controller.byobRequest?.respond(0);
         }
       },
-      cancel: (reason) => reader.cancel(reason)
+      cancel: (reason) => {
+        if (reader === undefined) reader = body.getReader();
+        return reader.cancel(reason);
+      }
     };
     bodyStream = new import_web3.ReadableStream(source);
     Object.defineProperty(bodyStream, kBodyStreamBrand, { value: true });